### PR TITLE
fix(pkg): pin-depends and git repos

### PR DESCRIPTION
**Pull request by rgrinberg**
Fix some bugs when resolving pinned packages

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

### CODE REVIEW

File: src/dune_pkg/pinned_package.ml [hunk 1]
```diff
@@ -79,16 +79,6 @@ let resolve_package loc name (url : OpamUrl.t) version =
       Resolved_package.local_fs package ~dir ~opam_file_path ~files_dir |> Fiber.return
     | `Git ->
       let* rev = Opam_repo.Source.of_opam_url loc url >>= Opam_repo.Source.rev in
-      let opam_file =
```
Comment (rgrinberg) [inline src/dune_pkg/pinned_package.ml:82]: This code was just hard coding `opam` to exist in the toplevel.

File: src/dune_pkg/pinned_package.ml [hunk 2]
```diff
@@ -106,17 +96,26 @@ let resolve_package loc name (url : OpamUrl.t) version =
              then `File
              else `Absent_or_unrecognized)
       in
-      let opam_file_path, files_dir = discover_layout ~stat in
+      let opam_file, files_dir = discover_layout ~stat in
```
Comment (rgrinberg) [inline src/dune_pkg/pinned_package.ml:99]: Now we correctly get the opam file from the layout we discovered

File: src/dune_pkg/pinned_package.ml [hunk 3]
```diff
@@ -106,17 +96,26 @@ let resolve_package loc name (url : OpamUrl.t) version =
...
       let+ opam_file_contents =
         (* CR-rgrinberg: not efficient to make such individual calls *)
-        Rev_store.At_rev.content rev opam_file_path
+        Rev_store.At_rev.content rev opam_file
         >>| function
         | Some p -> p
         | None ->
+          let files =
```
Comment (rgrinberg) [inline src/dune_pkg/pinned_package.ml:106]: This helped me debug the error so I left it

File: src/dune_pkg/pinned_package.ml [hunk 4]
```diff
@@ -55,7 +55,7 @@ let discover_layout loc ~stat name =
         | `Absent_or_unrecognized ->
           let file = Path.Local.relative opam_file_or_dir "opam" in
           (match stat file with
-           | `File -> opam_file, Some (Path.Local.relative opam_file_or_dir "files")
+           | `File -> file, Some (Path.Local.relative opam_file_or_dir "files")
```
Comment (rgrinberg) [inline src/dune_pkg/pinned_package.ml:58]: Another bug: the wrong file was being returned when we detect `opam/opam`

**Review by gridbugs — APPROVED** [review]

**Pull request merged**